docs/Extensibility.md

```diff
@@ -0,0 +1,104 @@
+
+To minimize risk, boilerplate, and avoid semantic clashes, we follow these rules:
+
+### Libraries
+
+- All function and storage variable names must be prefixed with the file name to prevent clashing with other libraries (e.g. `ERC20_approve` in the `ERC20` library)
+- Must not implement any `@external` or `@view` functions
+- Must not implement constructors
+- Must not call initializers on any function
+- Should implement initializers if needed (as any other library function, never as `@external`)
+
+### Contracts
+
+- Can import from libraries
+- Should implement `@external` functions if needed
+- Should implement a constructor that calls initializers
+- Must not call initializers on any function aside the constructor
+
+Note that since initializers will never be marked as `@external` and they won’t be called from anywhere but the contract constructor, there’s no risk of double-initialization. It’s up to the library developers not to make initializers inter-dependent to avoid weird dependency paths.
+
+## Presets
+
+Presets are pre-written contracts that extend from our library of contracts. They can be deployed as-is or used as templates for customization.
+
+Some presets are:
+
+- [Ownable](https://github.com/OpenZeppelin/cairo-contracts/blob/main/contracts/Ownable.cairo)
+- [ERC20](https://github.com/OpenZeppelin/cairo-contracts/blob/main/contracts/token/ERC20.cairo)
+  - [ERC20_Mintable](https://github.com/OpenZeppelin/cairo-contracts/blob/main/contracts/token/ERC20_Mintable.cairo)
+  - [ERC20_Pausable](https://github.com/OpenZeppelin/cairo-contracts/blob/main/contracts/token/ERC20_Pausable.cairo)
+- [ERC165](https://github.com/OpenZeppelin/cairo-contracts/blob/main/contracts/ERC165.cairo)
+- [ERC721](https://github.com/OpenZeppelin/cairo-contracts/blob/main/contracts/token/ERC721.cairo)
+  - [ERC721_Enumerable](https://github.com/OpenZeppelin/cairo-contracts/blob/main/contracts/token/ERC721_Enumerable.cairo)
+  - [ERC721_Mintable](https://github.com/OpenZeppelin/cairo-contracts/blob/main/contracts/token/ERC721_Mintable.cairo)
+  - [ERC721_Pausable](https://github.com/OpenZeppelin/cairo-contracts/blob/main/contracts/token/ERC721_Pausable.cairo)
+
+## Emulating hooks
+
+Unlike the Solidity version of [OpenZeppelin Contracts](https://github.com/OpenZeppelin/openzeppelin-contracts), this library does not implement [hooks](https://docs.openzeppelin.com/contracts/4.x/extending-contracts#using-hooks). The main reason being that Cairo does not support overriding functions.
+
+This is what a hook looks like in Solidity:
+
+```js
+abstract contract ERC20Pausable is ERC20, Pausable {
+    /**
+     * @dev See {ERC20-_beforeTokenTransfer}.
+     *
+     * Requirements:
+     *
+     * - the contract must not be paused.
+     */
```

Comment:
I feel like this doesn't add anything here.

```suggestion
```

Reply:
Agree it was a full snippet copypaste